lib/src/eval/compiler/expression/prefix.dart

```diff
@@ -19,11 +20,18 @@ const _opMap = {
 
 /// Compile a [PrefixExpression] to EVC bytecode
 Variable compilePrefixExpression(CompilerContext ctx, PrefixExpression e) {
-  var V = compileExpression(e.operand, ctx);
-
   final method = _opMap[e.operator.type] ??
       (throw CompileError('Unknown unary operator ${e.operator.type}'));
 
+  if ([TokenType.PLUS_PLUS, TokenType.MINUS_MINUS].contains(e.operator.type)) {
+    final V = compileExpressionAsReference(e.operand, ctx);
+    final L = V.getValue(ctx);
+    final type = V.resolveType(ctx);
```

Comment:
Rather than using `resolveType` here you should just use the type of `L` since you're already getting its value anyway and resolveType doesn't always work